src/Interpreters/parseColumnsListForTableFunction.cpp

```diff
@@ -44,8 +44,15 @@ bool tryParseColumnsListFromString(const std::string & structure, ColumnsDescrip
     if (!columns_list)
         return false;
 
-    columns = InterpreterCreateQuery::getColumnsDescription(*columns_list, context, false);
-    return true;
+    try
+    {
+        columns = InterpreterCreateQuery::getColumnsDescription(*columns_list, context, false);
+        return true;
+    }
+    catch (...)
+    {
+        return false;
```

Comment:
Why there can be an exception? Is some `tryLogCurrentException` needed? or at least `LOG_TRACE`?

Reply:
It can throw an exception when given `structure` is a valid columns declaration list from the parser's point of view, but contains invalid types. Examples: `x Tuple`, `x Decimal(999)`

Reply:
Then may be it makes sense to add `tryGetColumnsDescription` in this case?

Reply:
Good idea, I will check if it's possible

Reply:
I checked `getColumnsDescription` implementation and I don't think it's worth rewriting it to implement `tryGetColumnsDescription`, because it will make this function more complex and in some places it will require try/catch construction. Let's keep at as is.